_data/rules.yml

```diff
@@ -538,6 +538,50 @@ groups:
           - name: percona/mongodb_exporter
             doc_url: https://github.com/percona/mongodb_exporter
             rules:
+              - name: MongoDB replication lag
+                description: Mongodb replication lag is more than 10s
+                query: 'avg(mongodb_mongod_replset_member_optime_date{state="PRIMARY"}) - avg(mongodb_mongod_replset_member_optime_date{state="SECONDARY"})  > 10'
+                severity: critical
+              - name: MongoDB replication headroom
+                description: MongoDB replication headroom is <= 0
+                query: '(avg(mongodb_mongod_replset_oplog_tail_timestamp - mongodb_mongod_replset_oplog_head_timestamp) - (avg(mongodb_mongod_replset_member_optime_date{state="PRIMARY"}) - avg(mongodb_mongod_replset_member_optime_date{state="SECONDARY"}))) <= 0'
+                severity: critical
+              - name: MongoDB replication Status 3
+                description: MongoDB Replication set member either perform startup self-checks, or transition from completing a rollback or resync
+                query: "mongodb_mongod_replset_member_state == 3"
+                severity: critical
+              - name: MongoDB replication Status 6
+                description: MongoDB Replication set member as seen from another member of the set, is not yet known
+                query: "mongodb_mongod_replset_member_state == 6"
+                severity: critical
+              - name: MongoDB replication Status 8
+                description: MongoDB Replication set member as seen from another member of the set, is unreachable
+                query: "mongodb_mongod_replset_member_state == 8"
+                severity: critical
+              - name: MongoDB replication Status 9
+                description: MongoDB Replication set member is actively performing a rollback. Data is not available for reads
+                query: "mongodb_mongod_replset_member_state == 9"
+                severity: critical
+              - name: MongoDB replication Status 10
+                description: MongoDB Replication set member was once in a replica set but was subsequently removed
+                query: "mongodb_mongod_replset_member_state == 10"
+                severity: critical
+              - name: MongoDB number cursors open
+                description: Too many cursors opened by MongoDB for clients (> 10k)
+                query: 'mongodb_mongod_metrics_cursor_open{state="total"} > 10000'
+                severity: warning
+              - name: MongoDB cursors timeouts
+                description: Too many cursors are timing out
+                query: "increase(mongodb_mongod_metrics_cursor_timed_out_total[10m]) > 100"
+                severity: warning
+              - name: MongoDB too many connections
+                description: Too many connections
+                query: 'mongodb_connections{state="current"} > 500'
```

Comment:
Why is 500 a condition for an alert? By default mongo allows 64k connections, so alerting on less than 1% of allowed limit is a bit strange.